### PR TITLE
KEYCLOAK-15481 The new Account Console requires manage-account permis…

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.redhat.com/browse/KEYCLOAK-15481

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/error/KeycloakErrorHandler.java [hunk 1]
```diff
@@ -150,8 +150,20 @@ private RealmModel resolveRealm(KeycloakSession session) {
         attributes.put("url", new UrlBean(realm, theme, session.getContext().getUri().getBaseUri(), null));
         attributes.put("locale", new LocaleBean(realm, locale, session.getContext().getUri().getBaseUriBuilder(), messagesBundle));
 
+        String errorKey;
```
Comment (slaskawi) [inline services/src/main/java/org/keycloak/services/error/KeycloakErrorHandler.java:153]: This change is not essential for this commit. But without it, the `ForbiddenException` throws from `Auth#require(string)` end up being presented as an internal error (which is not great for the user experience).

**Comment by slaskawi** [comment]
>  I haven't yet checked the changes but do you think it'd make sense to add some simple test for this to the base-ui testsuite?

@vmuzikar I don't think so to be honest. This piece of code will be removed once we implement KEYCLOAK-15480. To me writing a test (and then removing it later) doesn't make much sense.

**Comment by vmuzikar** [comment]
@slaskawi I understood it's a permanent change. Say e.g. a user doesn't have access at all (not even ready only), then then this is necessary.

**Review by stianst — CHANGES_REQUESTED** [review]
This is not the correct approach - checking for the roles should be done on client-side in the new account console, not the server side to fetch a static html file. As I mentioned on the thread this is also not a workaround - the new account console should have checks in place to make sure users has the roles it needs, so that it can show a forbidden message if user does not have access. That would capability is needed in the future as well, it would only be the list of roles that changes.

**Comment by slaskawi** [comment]
> I understood it's a permanent change. Say e.g. a user doesn't have access at all (not even ready only), then this is necessary.

@vmuzikar I'm not exactly sure how [KEYCLOAK-15480|https://issues.redhat.com/browse/KEYCLOAK-15480] is going to be implemented but I imagine this piece needs to go away. Implementing granular permissions (e.g. the new Account Console in read-only mode) will require moving permission checks to react. This effectively means, the code I wrote will get deleted. Am I right @ssilvert ?

**Comment by vmuzikar** [comment]
@slaskawi I believe we agreed on closing this?

**Comment by slaskawi** [comment]
@vmuzikar Yep, closing.

**Pull request closed without merging**